interface.go

```diff
@@ -0,0 +1,184 @@
+	StorageSandboxSize      uint64      `json:",omitempty"` // Size in bytes that the container system drive should be expanded to if smaller
+	MemoryMaximumInMB       int64       `json:",omitempty"` // Maximum memory available to the container in Megabytes
+	HostName                string      // Hostname
+	MappedDirectories       []MappedDir // List of mapped directories (volumes/mounts)
+	SandboxPath             string      // Location of unmounted sandbox (used for Hyper-V containers)
+	HvPartition             bool        // True if it a Hyper-V Container
+	EndpointList            []string    // List of networking endpoints to be attached to container
+}
+
+const (
+	NotificationType_None           string = "None"
+	NotificationType_GracefulExit   string = "GracefulExit"
+	NotificationType_ForcedExit     string = "ForcedExit"
+	NotificationType_UnexpectedExit string = "UnexpectedExit"
+	NotificationType_Reboot         string = "Reboot"
+	NotificationType_Constructed    string = "Constructed"
+	NotificationType_Started        string = "Started"
+	NotificationType_Paused         string = "Paused"
+	NotificationType_Unknown        string = "Unknown"
+)
+
+type ContainerProperties struct {
+	Id                string
+	Name              string
+	SystemType        string
+	Owner             string
+	SiloGuid          string `json:",omitempty"`
+	IsDummy           bool   `json:",omitempty"`
+	RuntimeId         string `json:",omitempty"`
+	Stopped           bool   `json:",omitempty"`
+	ExitType          string `json:",omitempty"`
+	AreUpdatesPending bool   `json:",omitempty"`
+}
+
+// Container represents a created (but not necessarily running) container.
+type Container interface {
+	// Start synchronously starts the container.
+	Start() error
+
+	// Shutdown requests a container shutdown, but it may not actually be shutdown until Wait() succeeds.
+	Shutdown() error
+
+	// Terminate requests a container terminate, but it may not actually be terminated until Wait() succeeds.
+	Terminate() error
+
+	// Waits synchronously waits for the container to shutdown or terminate.
+	Wait() error
+
+	// WaitTimeout synchronously waits for the container to terminate or the duration to elapse. It
+	// returns false if timeout occurs.
+	WaitTimeout(time.Duration) (bool, error)
+
+	// Pause pauses the execution of a container.
+	Pause() error
+
+	// Resume resumes the execution of a container.
+	Resume() error
+
+	// Properties returns the properties of the container
+	Properties() (*ContainerProperties, error)
+
+	// CreateProcess launches a new process within the container.
+	CreateProcess(c *ProcessConfig) (Process, error)
+
+	// OpenProcess gets an interface to an existing process within the container.
+	OpenProcess(pid int) (Process, error)
+
+	// Close cleans up any state associated with the container but does not terminate or wait for it.
+	Close() error
+}
```

Comment:
I did see some mention of some stats. I'll talk to Lars and implement them if so.